Domains: Add notice when canceling or transferring a free Gravatar domain

leonardost opened the pull request:
## Proposed Changes

This PR adds notices and warnings explaining that users won't be able to get another free Gravatar domain in these situations:
- During checkout for a free Gravatar domain (via the `/start/domain-for-gravatar` flow)
- When transferring a free Gravatar domain away and
- When deleting a free Gravatar domain

### Screenshots

| Page | Before | After |
| --- | --- | --- |
| Checkout | ![Markup on 2024-06-14 at 20:35:50](https://github.com/Automattic/wp-calypso/assets/5324818/23086fe7-e92e-45c8-a6a7-8bdfa01d9b5d) | ![Markup on 2024-06-17 at 15:34:15](https://github.com/Automattic/wp-calypso/assets/5324818/985d1468-08bc-453a-945f-23e0b97893a6) |
| Outbound transfer | ![Screenshot 2024-06-13 at 18 24 15](https://github.com/Automattic/wp-calypso/assets/5324818/e588894f-179c-40ad-8668-f32bc5d53b19) | ![Screenshot 2024-06-17 at 15 35 31](https://github.com/Automattic/wp-calypso/assets/5324818/4e43b0b1-2f8e-493f-9877-19d40dd32a60) |
| Disable domain auto-renew | ![Screenshot 2024-06-13 at 16 05 19](https://github.com/Automattic/wp-calypso/assets/5324818/a0970a5c-3994-48d1-a1a8-21c0dba35f7b) | ![Markup on 2024-06-17 at 15:36:01](https://github.com/Automattic/wp-calypso/assets/5324818/e01f765b-ecdd-45b8-8b90-2f9dca0a9f38) |
| Delete domain (after turning auto-renew off) | ![Screenshot 2024-06-13 at 16 12 29](https://github.com/Automattic/wp-calypso/assets/5324818/d5c47668-51bb-49b8-b7b2-0b0fc4e8948e) | ![Markup on 2024-06-17 at 15:36:29](https://github.com/Automattic/wp-calypso/assets/5324818/7c553077-53cc-49b0-9e8e-633576be78d5) |

## Why are these changes being made?

- This is part of the Custom domains on Gravatar project (pcYYhz-24z-p2).

## Testing Instructions

- Open the live Calypso link or build this branch locally
- Checkout
  - Go to the domains for Gravatar flow (`/start/domain-for-gravatar`) and add a domain to the cart
  - Ensure the explanation about only one free Gravatar domain is shown in the checkout page
  - Visit the other flows (`/start`, adding domains to an existing site, etc) and ensure the free Gravatar domain copy isn't shown in the checkout page
- Outbound transfer
  - When you already have a free Gravatar domain, go to its domain transfer page and ensure a warning notice is shown
- Cancel domain
  - When you already have a free Gravatar domain, go to the domain cancelation page and ensure a warning text is shown
  - After you disable auto-renew for that domain, try to cancel it and ensure the deletion confirmation dialog also contains warning text explaining that you won't be able to get another free Gravatar domain

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

leonardost committed:
Add notice when canceling or transferring a free Gravatar domain

leonardost committed:
Show copy about free Gravatar domain in checkout

leonardost committed:
Remove text specific to canceling Gravatar domains in AGP (add grace period)

Free Gravatar domains are not refundable since there's no non-zero
transaction associated with them. Because of that, the AGP specific
copy is not shown in the purchase cancelation page.

jjchrisdiehl commented on client/me/purchases/remove-purchase/remove-domain-dialog/index.jsx:
```diff
@@ -32,7 +32,7 @@ class RemoveDomainDialog extends Component {
 
 	renderDomainDeletionWarning( productName ) {
-		const { translate, slug, currentRoute } = this.props;
+		const { translate, slug, currentRoute, isGravatarDOmain } = this.props;
```
`isGravatarDOmain` typo?

jjchrisdiehl commented on client/me/purchases/remove-purchase/remove-domain-dialog/index.jsx:
```diff
@@ -44,6 +44,13 @@ class RemoveDomainDialog extends Component {
 						}
 					) }
 				</p>
+				{ isGravatarDOmain && (
```
Typo here too

jjchrisdiehl commented on client/me/purchases/remove-purchase/remove-domain-dialog/index.jsx:
```diff
@@ -267,6 +274,7 @@ export default connect( ( state, ownProps ) => {
 	const selectedDomainName = getName( ownProps.purchase );
 	const selectedDomain = getSelectedDomain( { domains, selectedDomainName } );
 	return {
+		isGravatarDOmain: selectedDomain.isGravatarDomain,
```
Here too

jjchrisdiehl commented:
Overall looks good to me - only noted a typo that should be fixed 👍

hambai reviewed (commented):
So besides the typo my concern here is the "free Gravatar domain" in the copy. Would it make sense to change that to "your complimentary Gravatar profile domain" or "free domain for your Gravatar profile"?

**So in the checkout:**
> This is a free Gravatar domain. You can only get one free domain to use with Gravatar

would become

> This is a complimentary Gravatar profile domain. You're eligible to only one free domain for your Gravatar profile.

**In outbound transfer:**

> This is a free Gravatar domain. If you transfer it to another registrar, you won't be able to get another free domain to use with Gravatar.

would become

> This is a complimentary Gravatar profile domain. You won't be able to get another free domain in case you transfer this one to another registrar.

**In Disable domain auto-renew:**

> This is a free Gravatar domain. If it expires, you won't be able to get another free domain to use with Gravatar.

would become

> This is a complimentary Gravatar profile domain. If it expires, you won't be able to get another free domain to use for your Gravatar profile

**And in Delete domain:**

> Also, this is a free Gravatar domain. If if is deleted, you won't be able to get another free domain to use with Gravatar.

would become

> Also, this is a complimentary Gravatar profile domain. If if is deleted, you won't be able to get another free domain to use with your Gravatar profile.

I'd prefer to get a second look by a native speaker though

rafaelgallani commented on client/me/purchases/remove-purchase/remove-domain-dialog/index.jsx:
```diff
@@ -44,6 +44,13 @@ class RemoveDomainDialog extends Component {
+				{ isGravatarDOmain && (
+					<p>
+						{ translate(
+							"Also, this is a free Gravatar domain. If if is deleted, you won't be able to get another free domain to use with Gravatar."
```
Small typo
```suggestion
							"Also, this is a free Gravatar domain. If it is deleted, you won't be able to get another free domain to use with Gravatar."
```

rafaelgallani commented on client/my-sites/checkout/src/components/domain-registration-free-gravatar-domain.tsx:
```diff
@@ -0,0 +1,27 @@
+	return (
+		<CheckoutTermsItem>
+			{ translate(
+				'This is a free Gravatar domain. You can only get one free domain to use with Gravatar'
```
Maybe add a period to match all other checkout messages/terms?
```suggestion
				'This is a free Gravatar domain. You can only get one free domain to use with Gravatar.'
```

rafaelgallani reviewed (commented):
Just some small suggestions

rafaelgallani commented:
> So besides the typo my concern here is the "free Gravatar domain" in the copy. Would it make sense to change that to "your complimentary Gravatar profile domain" or "free domain for your Gravatar profile"?

I'm not a native speaker, but I agree with Kamen's suggestion, although I prefer the latter — it feels more natural, and the translation seems better... And we still stand out the fact that the domain was given for free.

leonardost committed:
Fix typo in `isGravatarDOmain`

leonardost committed:
Add period at end of sentence

leonardost committed:
Update copy according to code review suggestions

leonardost commented:
Thanks for the review @jjchrisdiehl, @hambai and @rafaelgallani! I've fixed the typo and updated the copy with your suggestions. I'll ask @delputnam for a native speaker review 🙂

delputnam commented on client/me/purchases/cancel-purchase/refund-information.jsx:
```diff
@@ -287,10 +290,20 @@ const CancelPurchaseRefundInformation = ( {
+		if ( isGravatarDomain ) {
+			text.push(
+				i18n.translate(
+					"This is a free domain for your Gravatar profile. If it expires, you won't be able to get another free domain to use for your Gravatar profile."
```
This is a bit wordy, but let me know what you think:

> This domain, for use with your Gravatar profile, is provided at no cost for the first year. Please be aware that this offer is limited to one free domain per user. If you cancel this domain, you will have to pay the regular price to register another domain.

delputnam commented on client/me/purchases/remove-purchase/remove-domain-dialog/index.jsx:
```diff
@@ -44,6 +44,13 @@ class RemoveDomainDialog extends Component {
+				{ isGravatarDomain && (
+					<p>
+						{ translate(
+							"Also, this is a free domain for your Gravatar profile. If it is deleted, you won't be able to get another free domain to use with your Gravatar profile."
```
How about this?

This domain is provided at no cost for the first year for use with your Gravatar profile. This offer is limited to one per user. If you cancel this domain, you will have to pay the standard price to register another domain for your Gravatar profile.

delputnam commented on client/my-sites/checkout/src/components/domain-registration-free-gravatar-domain.tsx:
```diff
@@ -0,0 +1,27 @@
+	return (
+		<CheckoutTermsItem>
+			{ translate(
+				"This is a free domain for your Gravatar profile. You're eligible to only one free domain for your Gravatar profile."
```
> This domain is provided at no cost for the first year for use with your Gravatar profile. This offer is limited to one per user. If you cancel this domain, you will have to pay the standard price to register another domain for your Gravatar profile.

delputnam commented on client/my-sites/domains/domain-management/transfer/transfer-page/index.tsx:
```diff
@@ -372,6 +373,14 @@ const TransferPage = ( props: TransferPageProps ) => {
+					<InfoNotice
+						redesigned
+						text={ __(
+							"This is a free domain for your Gravatar profile. If you transfer it to another registrar, you won't be able to get another free domain to use with your Gravatar profile."
```
> This domain is provided at no cost for the first year for use with your Gravatar profile. This offer is limited to one per user. If you transfer it to another registrar, you will have to pay the standard price to register another domain for your Gravatar profile.

delputnam reviewed (approved):
Let's try the language I've suggested and see how that looks.

leonardost committed:
Update copy according to code review suggestions
